src/pages/database/RenameNotePopup.tsx

```diff
@@ -0,0 +1,48 @@
+import { Form, Menu, Popup } from 'semantic-ui-react';
+
+import { setEol } from '../../utils/helpers';
+import useStateCallback from '../../utils/useStateCallback';
+import { RenameNoteProps } from './Database';
+
+const RenameNotePopup: React.FC<RenameNoteProps> = props => {
+  const [open, setOpen] = useStateCallback(false);
+
+  // Updates title after enter key pressed in popup form
+  const onEnterKey = (noteId: string) => (e: React.KeyboardEvent<HTMLDivElement>) => {
+    if (e.key === 'Enter') {
+      props.updateNoteTitleHandler(
+        noteId,
+        (document.getElementById(noteId + 'rename') as HTMLInputElement).value
+      );
+      setOpen(false);
+    }
+  };
+
+  // Callback to set line after popup opens
+  const setLine = () => {
+    setEol(document.getElementById(props.note.id + 'rename') as HTMLElement);
+  };
+
+  return (
+    <Popup
+      on="click"
+      open={open}
+      onOpen={() => setOpen(true, setLine)}
+      onClose={() => setOpen(false)}
+      basic
+      pinned
+      trigger={<Menu.Item onClick={() => {}}>Rename Note </Menu.Item>}
+    >
+      <Form>
+        <input
+          id={props.note.id + 'rename'}
+          onKeyDown={onEnterKey(props.note.id)}
+          type="text"
+          defaultValue={props.note.title}
+        />
+      </Form>
+    </Popup>
```

Comment:
same changes needed as in RenameDatabasePopup.tsx.

infact if the 2 functionalities are going to be the same, and UI-wise both are going to have the same 3 buttons, then we can generalise this component into a common component, and simply pass different handlers and names down